osrf/mbzirc
# Fixed wing controller.

This PR adds a fixed wing controller for the zephyr. It implements a simple attitude controller that uses the `mavros_msgs` `AttitudeTarget.msg` for being commanded. The zephyr is a bit of an oddity when it comes to fix wings as it doesn't have control surfaces for controlling yaw. Instead one can get the aircraft to roll in one direction and thus achieve yaw control. I believe we can add this feature in a bit later as its a relatively trivial feature to add.

## To Test
:warning: This PR pulls in a new dependency on `mavros_msgs` you are likely going to have to do the whole `rosdep` drill again.

Start an empty ignition gazebo world in one window:
```
ign gazebo
```
Spawn a fixed wing zephyr in a separate terminal:
```
ros2 launch mbzirc_ign spawn.launch.py name:=zephyr world:=empty_platform model:=mbzirc_fixed_wing type:=uav x:=0 y:=0 z:=4
```
Publish a `mavros_msgs/msg/AttitudeTarget` on `/zephyr/cmd/attitude` with a `thrust` of 1000.
![image](https://user-images.githubusercontent.com/542272/155465165-cab11842-e5ae-49b6-b4ff-5582bc2c73d6.png)

Start the simulation and you should see the plane try to level out:
![plane_leveling_out](https://user-images.githubusercontent.com/542272/155466621-cc9f4cec-039a-413a-97d1-43dde18d5dc4.gif)

## TODO
* [x] Document plugin
